lib/src/ctx/context.rs

```diff
@@ -91,6 +100,31 @@ impl<'a> Context<'a> {
 		}
 	}
 
+	pub fn with_live_sess(mut self, session: &Session) -> Self {
+		self.add_value(SESSION_KEY, Value::None);
+		match session.sd.clone() {
+			None => self.add_value(AUTH_KEY, Value::Null),
+			Some(s) => self.add_value(AUTH_KEY, s),
+		}
+		match session.sc.clone() {
```

Comment:
Same question here to borrow the session parameter, please resolve if the above is answered

Reply:
In this case, clone is unnecessary even in the `Some` branch because `Value::from(s.as_str())` will still result in the same thing.

Reply:
We can still have a clone that uses copy I believe. Tracked here https://github.com/surrealdb/surrealdb/issues/2829

Reply:
Even if we implemented that feature request, it wouldn't solve this. `session.sc` is `Option<String>` not `Option<Session>` so making `Session` copy wouldn't help. This can still be written as

```rust
match &session.tk {
    Some(scope) => self.add_value(TOKEN_KEY, Value::from(scope.as_str())),
    None => self.add_value(TOKEN_KEY, Value::Null),
}
```

Reply:
Yes, resolved now thank you